fastlane/spec/actions_specs/import_from_git_spec.rb

```diff
@@ -208,26 +209,53 @@ def self.is_supported?(platform)
           end").runner.execute(:test)
         end
 
-        # Because we don't know if the specified value represents a branch or a
-        # tag. And knowing that it doesn't make sense to cache a branch, that
-        # points to a different commit when one is added, as opposed to a tag,
-        # we decided to enable caching only when used with the `version` param.
-        it "ignores it when tag is specified via branch param" do
-          expect(UI).not_to receive(:message).with(caching_message)
-          expect(UI).to receive(:important).with('Works since v1')
+        # Meaning, `git fetch` and `git merge` are performed when caching is eligible
+        # and the version isn't specified in the local clone.
+        it "works with updated branch" do
+          Dir.chdir(source_directory_path) do
+            `git checkout "version-2.1" 2>&1`
+            File.write('fastlane/Fastfile', <<-FASTFILE)
+              lane :works do
+                UI.important('Works until v5')
+              end
+            FASTFILE
+            `git add .`
+            `git commit --message "Version 5"`
+          end
+
+          allow(UI).to receive(:message)
+          expect(UI).to receive(:message).with(caching_message)
+          expect(UI).to receive(:important).with('Works until v5')
 
           Fastlane::FastFile.new.parse("lane :test do
-            import_from_git(url: '#{source_directory_path}', branch: '3', cache_path: '#{cache_directory_path}')
+            import_from_git(url: '#{source_directory_path}', branch: 'version-2.1', cache_path: '#{cache_directory_path}')
 
             works
           end").runner.execute(:test)
         end
 
-        it "ignores it when version is not specified" do
-          expect(UI).not_to receive(:message).with(caching_message)
+        # Respects the specified version even if the branch is specified
+        it "works with new tags and branch" do
+          Dir.chdir(source_directory_path) do
+            `git checkout "master" 2>&1`
+            File.write('fastlane/Fastfile', <<-FASTFILE)
+              lane :works do
+                UI.important('Works until v6')
+              end
+            FASTFILE
+            `git add .`
+            `git commit --message "Version 6"`
+            `git tag "6"`
+          end
+
+          allow(UI).to receive(:message)
+          expect(UI).to receive(:message).with(caching_message)
+          expect(UI).to receive(:important).with('Works until v6')
 
           Fastlane::FastFile.new.parse("lane :test do
-            import_from_git(url: '#{source_directory_path}', cache_path: '#{cache_directory_path}')
+            import_from_git(url: '#{source_directory_path}', branch: 'version-2.1', version: '6', cache_path: '#{cache_directory_path}')
```

Comment:
So the `branch` is essentially ignored in this case?

Reply:
Yes, that's correct. This is important because
- make sure that it won't break the current cache system with this change.
- specifying `version` should be prioritized over `branch` given that it's an optional parameter and optimistic match operator or multiple conditions can be used.

Reply:
I see, thanks!